README.md

```diff
@@ -1,47 +1,184 @@
-# cosmian_crypto_core &emsp; [![Build Status]][actions] [![Latest Version]][crates.io]
+# Cosmian CryptoCore &emsp; [![Build Status]][actions] [![Latest Version]][crates.io]
 
 
 [Build Status]: https://img.shields.io/github/workflow/status/Cosmian/crypto_core/CI%20checks/main
 [actions]: https://github.com/Cosmian/crypto_core/actions?query=branch%3Amain
 [Latest Version]: https://img.shields.io/crates/v/cosmian_crypto_core.svg
 [crates.io]: https://crates.io/crates/cosmian_crypto_core
 
-This crate implements crypto primitives which are used in many other
-Cosmian cryptographic resources:
+This crate implements the cryptographic primitives which are used in many other
+Cosmian resources:
 
-- symmetric cryptography primitives can be found in the `symmetric_crypto` module;
-- asymmetric cryptography primitives can be found in the `asymmetric_crypto` module;
-- a Random Number Generator (RNG) can be found in the `entropy` module.
+- a Data Encryption Method (DEM) based on AES256-GCM (128 bytes of security)
+  offering a simple API (Nonce management and authentication checks are internals);
+- a Diffie-Hellman key pair based on the curve25519 (128 bytes of security)
+  allowing implementation of Diffie-Hellman based asymmetric cryptography;
+- a Random Number Generator (RNG) implementing the ChaCha algorithm with 12
+  rounds (128 bytes of security).
+- a Key Derivation Function (KDF) based on SHAKE128 (128 bytes of security);
 
-The crate also defines `CryptoCoreError`, the error type, and a few traits.
+All these primitives are use pure rust implementations, are secure, and used
+the fastest known algorithms. They offer a great basis on which to build more
+complex softwares.
 
-## Symmetric Crypto
+## Getting started
 
-This crate implements a Data Encapsulation Mechanism (DEM) based on the AES 256
-GCM algorithm, as described in the [ISO 2004](https://www.shoup.net/iso/std6.pdf).
-This implementation is 128-bits secure.
+The following example can be run using `cargo run --example demo`.
 
-It uses the [`aes_gcm`](https://docs.rs/aes-gcm/latest/aes_gcm/index.html)
-implementation of the AES GCM algorithm. This implementation makes use of the
-AES instruction set when available, which allows for a high encryption speed.
+```
```

Comment:
I think in Markdown (at least with Github rendering), you can include source code file: https://github.com/marketplace/actions/markdown-embed-code-from-file#how-to-use